mc/CodeAnalysis/Lexer.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections.Generic;
+
+namespace Minsk.CodeAnalysis
+{
+    class Lexer
+    {
+        private readonly string _text;
+        private int _position;
+        private List<string> _diagnostics = new List<string>();
+
+        public Lexer(string text)
+        {
+            _text = text;
+        }
+
+        public IEnumerable<string> Diagnostics => _diagnostics;
+
+        private char Current
+        {
+            get
+            {
+                if (_position >= _text.Length)
+                    return '\0';
+
+                return _text[_position];
+            }
+        }
+
+        private void Next()
+        {
+            _position++;
+        }
+
+        public SyntaxToken NextToken()
+        {
+            // <numbers>
+            // + - * / ( )
+            // <whitespace>
+
+            if (_position >= _text.Length)
+                return new SyntaxToken(SyntaxKind.EndOfFileToken, _position, "\0", null);
+
+            if (char.IsDigit(Current))
+            {
+                var start = _position;
+
+                while (char.IsDigit(Current))
+                    Next();
+
+                var length = _position - start;
+                var text = _text.Substring(start, length);
+                if (!int.TryParse(text, out var value))
+                    _diagnostics.Add($"The number {_text} isn't valid Int32.");
+
+                return new SyntaxToken(SyntaxKind.NumberToken, start, text, value);
+            }
+
+            if (char.IsWhiteSpace(Current))
+            {
+                var start = _position;
+
+                while (char.IsWhiteSpace(Current))
+                    Next();
+
+                var length = _position - start;
+                var text = _text.Substring(start, length);
+                return new SyntaxToken(SyntaxKind.WhitespaceToken, start, text, null);
+            }
+
+            if (Current == '+')
```

Comment:
Wouldn't it be better to use a dictionary with SyntaxToken by not numeric symbol?